src/error.rs

```diff
@@ -0,0 +1,35 @@
+use std::io;
+
+use thiserror::Error as ThisError;
+
+pub type Result<T> = core::result::Result<T, Error>;
+
+#[derive(ThisError, Debug)]
+pub enum Error {
+    #[error("IO error {0}")]
+    IO(#[from] io::Error),
+    #[error("Datafusion error: {0}")]
+    Datafusion(#[from] datafusion::error::DataFusionError),
```

Comment:
If we allow any error to be **implicitly** (via `?`) converted into `odata::Error` - then this is no different to returning `Box<dyn Error>` or `anyhow` - don't think this is a good practice, especially in libraries.

While getting "table not found" error from DataFusion in one context may be equivalent to "collection not found" and warrant 404 in one context, in another context it may not have the same meaning and should be treated as internal error rather than 404. Making all datafusion errors implicitly convertable into our `Error` makes it very easy to just put a `?` everywhere without thinking about the context.